libdrgn/dwarf_info_cache.c

```diff
@@ -596,17 +637,46 @@ drgn_compound_type_from_dwarf(struct drgn_dwarf_info_cache *dicache,
 	type = malloc(sizeof(*type));
 	if (!type)
 		return &drgn_enomem;
+	drgn_template_parameter_vector_init(&templates);
+
+	r = dwarf_child(die, &child);
+	while (r == 0) {
+		if (dwarf_tag(&child) == DW_TAG_template_type_parameter) {
```

Comment:
Since implementing support for incomplete types was a little sensitive as I moved stuff around, I made it into its own commit for now. If it looks good, I'll squash it into the others where they belong. 

Right now, I ended up doing two passes over the children, to avoid creating the members array for incomplete types. However I could also do a single pass, and delete any (erroneous) member entries for incomplete types. Let me know what you like more!

Reply:
We can keep it as one pass by modifying the loop over the DWARF children to check `if (!declaration && dwarf_tag(&child) == DW_TAG_member)`. You'll also have to move the incomplete type initialization to happen in the same place as the complete types and skip the `dwarf_bytesize()` for incomplete types